cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -58,12 +58,13 @@
   /**
    * Logs start of program run.
    *
-   * @param id        Info about program
-   * @param pid       run id
-   * @param startTime start timestamp in seconds; if run id is time-based pass the time from the run id
-   * @param adapter   name of the adapter associated with the run
+   * @param id         Info about program
+   * @param pid        run id
+   * @param startTime  start timestamp in seconds; if run id is time-based pass the time from the run id
+   * @param adapter    name of the adapter associated with the run
+   * @param twillRunId twill run id
    */
-  void setStart(Id.Program id, String pid, long startTime, String adapter);
+  void setStart(Id.Program id, String pid, long startTime, String adapter, @Nullable String twillRunId);
```

Comment:
Shouldn't just update the method (as that would introduce unnecessary changes to pass in `null`). Should just add an overloaded method and have the existing one modified to call this one with `null`.